uvtools/dspec.py

```diff
@@ -194,20 +201,34 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
             pass
         else:
             raise ValueError("Didn't recognize filt2d_mode {}".format(filt2d_mode))
-            
+
         # run clean
         _d_cl, info = aipy.deconv.clean(_d, _w, area=area, tol=tol, stop_if_div=False, maxiter=maxiter, gain=gain)
-        d_mdl = np.fft.fft2(_d_cl + info['res'] * area, axes=(0, 1))
+        _d_res = info['res']
         del info['res']
 
-    d_res = data - d_mdl
+    # add resid to model in CLEAN bounds
+    if add_clean_residual:
+        _d_cl += _d_res * area
+        _d_res -= _d_res * area
+
+    # fft back to input space
+    if clean2d:
+        d_mdl = np.fft.fft2(_d_cl, axes=(0, 1))
+        d_res = np.fft.fft2(_d_res, axes=(0, 1))
+    else:
+        d_mdl = np.fft.fft(_d_cl)
+        d_res = np.fft.fft(_d_res)
 
-    return d_mdl, d_res, info
+    # form clean data
+    d_cln = d_mdl + d_res
```

Comment:
I'd advocate removing this, but if you don't having `_d_cl` FFT to `d_mdl` while you also have a `d_cln` is very confusing. If you remove `d_cln` then it doesn't matter.